### PR TITLE
Update dynamic plan actions

### LINKED ISSUE #1168: Update Dynamic Plans Action templates
The current expressions for various templates are wrong. Update the plan templates used to generate Plans as per below

IRS
```
  "action": [
            {
                "identifier": "b996adc3-574d-5bc0-903d-d00d38252a9f",
                "prefix": 1,
                "title": "Spray Structures",
                "description": "Visit each structure in the operational area and attempt to spray",
                "code": "IRS",
                "timingPeriod": {
                    "start": "2020-08-17",
                    "end": "2020-08-24"
                },
                "reason": "Routine",
                "goalId": "IRS",
                "subjectCodableConcept": {
                    "text": "Location"
                },
                "trigger": [
                    {
                        "type": "named-event",
                        "name": "plan-activation"
                    },
                    {
                        "type": "named-event",
                        "name": "event-submission",
                        "expression": {
                            "description": "Trigger when a Register_Structure event is submitted",
                            "expression": "questionnaire = 'Register_Structure'"
                        }
                    }
                ],
                "condition": [
                    {
                        "kind": "applicability",
                        "expression": {
                            "description": "Structure is residential or type does not exist",
                            "expression": "$this.is(FHIR.QuestionnaireResponse) or $this.type.where(id='locationType').exists().not() or $this.type.where(id='locationType').text = 'Residential Structure'"
                        }
                    },
                    {
                        "kind": "applicability",
                        "expression": {
                            "description": "Apply to residential structures in Register_Structure questionnaires",
                            "expression": "$this.is(FHIR.Location) or (questionnaire = 'Register_Structure' and $this.item.where(linkId='structureType').answer.value ='Residential Structure')"
                        }
                    }
                ],
                "definitionUri": "spray_form.json",
                "type": "create"
            }
        ]
```

MDA 

```
  "action": [
    {
      "identifier": "79e9ad73-45a0-52b9-82e9-50d21c29bb78",
      "prefix": 1,
      "title": "Family Registration",
      "description": "Register all families &amp; family members in all residential structures enumerated (100%) within the operational area",
      "code": "RACD Register Family",
      "timingPeriod": {
        "start": "2020-08-18",
        "end": "2020-08-25"
      },
      "reason": "Routine",
      "goalId": "RACD_register_families",
      "subjectCodableConcept": {
        "text": "Location"
      },
      "trigger": [
        {
          "type": "named-event",
          "name": "plan-activation"
        },
        {
          "type": "named-event",
          "name": "event-submission",
          "expression": {
            "description": "Trigger when a Register_Structure event is submitted",
            "expression": "questionnaire = 'Register_Structure'"
          }
        }
      ],
      "condition": [
        {
          "kind": "applicability",
          "subjectCodableConcept": {
            "text": "Family"
          },
          "expression": {
            "description": "Structure is residential or type does not exist and family does not exist",
            "expression": "$this.is(FHIR.QuestionnaireResponse) or (($this.type.where(id='locationType').exists().not() or $this.type.where(id='locationType').text = 'Residential Structure') and $this.contained.exists().not())"
          }
        },
        {
          "kind": "applicability",
          "expression": {
            "description": "Apply to residential structures in Register_Structure questionnaires",
            "expression": "$this.is(FHIR.Location) or (questionnaire = 'Register_Structure' and $this.item.where(linkId='structureType').answer.value ='Residential Structure')"
          }
        }
      ],
      "definitionUri": "family_register.json",
      "type": "create"
    },
    {
      "identifier": "69dd5115-a8fc-5b06-b3a1-75b7e303a7f9",
      "prefix": 2,
      "title": "Distribute Drugs",
      "description": "Visit all residential structures (100%) and dispense prophylaxis to each registered person",
      "code": "MDA Dispense",
      "timingPeriod": {
        "start": "2020-08-18",
        "end": "2020-08-25"
      },
      "reason": "Routine",
      "goalId": "MDA_Dispense",
      "subjectCodableConcept": {
        "text": "Person"
      },
      "trigger": [
        {
          "type": "named-event",
          "name": "plan-activation"
        },
        {
          "type": "named-event",
          "name": "event-submission",
          "expression": {
            "description": "Trigger when a Family Registration or Family Member Registration event is submitted",
            "expression": "questionnaire = 'Family_Registration' or questionnaire = 'Family_Member_Registration'"
          }
        }
      ],
      "condition": [
        {
          "kind": "applicability",
          "expression": {
            "description": "Person or person associated with questionaire response is older than 5 years and younger than 15 years",
            "expression": "($this.is(FHIR.Patient) and $this.birthDate <= today() - 5 'years' and $this.birthDate > today() - 15 'years') or ($this.contained.where(Patient.birthDate <= today() - 5 'years' and $this.birthDate > today() - 15 'years').exists())"
          }
        }
      ],
      "definitionUri": "mda_dispense.json",
      "type": "create"
    },
    {
      "identifier": "f4d7243b-4b6d-532c-89dc-6d593ee46988",
      "prefix": 3,
      "title": "MDA Adherence",
      "description": "Visit all residential structures (100%) and confirm adherence of each registered person",
      "code": "MDA Adherence",
      "timingPeriod": {
        "start": "2020-08-18",
        "end": "2020-08-25"
      },
      "reason": "Routine",
      "goalId": "MDA_Adherence",
      "subjectCodableConcept": {
        "text": "Person"
      },
      "trigger": [
        {
          "type": "named-event",
          "name": "event-submission",
          "expression": {
            "description": "Trigger when a MDA Dispense event is submitted",
            "expression": "questionnaire = 'mda_dispense'"
          }
        }
      ],
      "condition": [
        {
          "kind": "applicability",
          "expression": {
            "description": "The person fully received the dispense activity",
            "expression": "$this.item.where(linkId='business_status').answer.value = 'Fully Received'"
          }
        }
      ],
      "definitionUri": "mda_adherence.json",
      "type": "create"
    }
  ]
```

FI

```
"action": [
    {
      "identifier": "5bc2b863-9127-5b5b-a7ee-7a52775576e7",
      "prefix": 1,
      "title": "Family Registration",
      "description": "Register all families &amp; family members in all residential structures enumerated (100%) within the operational area",
      "code": "RACD Register Family",
      "timingPeriod": {
        "start": "2020-08-13",
        "end": "2020-08-20"
      },
      "reason": "Routine",
      "goalId": "RACD_register_families",
      "subjectCodableConcept": {
        "text": "Location"
      },
      "trigger": [
        {
          "type": "named-event",
          "name": "plan-activation"
        },
        {
          "type": "named-event",
          "name": "event-submission",
          "expression": {
            "description": "Trigger when a Register_Structure event is submitted",
            "expression": "questionnaire = 'Register_Structure'"
          }
        }
      ],
      "condition": [
        {
          "kind": "applicability",
          "subjectCodableConcept": {
            "text": "Family"
          },
          "expression": {
            "description": "Structure is residential or type does not exist and Family does not exist",
            "expression": "$this.is(FHIR.QuestionnaireResponse) or (($this.type.where(id='locationType').exists().not() or $this.type.where(id='locationType').text = 'Residential Structure') and $this.contained.exists().not())"
          }
        },
        {
          "kind": "applicability",
          "expression": {
            "description": "Apply to residential structures in Register_Structure questionnaires",
            "expression": "$this.is(FHIR.Location)  or (questionnaire = 'Register_Structure' and $this.item.where(linkId='structureType').answer.value ='Residential Structure')"
          }
        }
      ],
      "definitionUri": "family_register.json",
      "type": "create"
    },
    {
      "identifier": "28e7da08-aad7-5c4a-87ad-e955e54bb324",
      "prefix": 2,
      "title": "Blood Screening",
      "description": "Visit all residential structures (100%) within a 1 km radius of a confirmed index case and test each registered person",
      "code": "Blood Screening",
      "timingPeriod": {
        "start": "2020-08-13",
        "end": "2020-08-20"
      },
      "reason": "Investigation",
      "goalId": "RACD_Blood_Screening",
      "subjectCodableConcept": {
        "text": "Person"
      },
      "trigger": [
        {
          "type": "named-event",
          "name": "plan-activation"
        },
        {
          "type": "named-event",
          "name": "event-submission",
          "expression": {
            "description": "Trigger when a Family Registration or Family Member Registration event is submitted",
            "expression": "questionnaire = 'Family_Registration' or questionnaire = 'Family_Member_Registration'"
          }
        }
      ],
      "condition": [
        {
          "kind": "applicability",
          "expression": {
            "description": "Person is older than 5 years or person associated with questionnaire response if older than 5 years",
            "expression": "($this.is(FHIR.Patient) and $this.birthDate <= today() - 5 'years') or ($this.contained.where(Patient.birthDate <= today() - 5 'years').exists())"
          }
        }
      ],
      "definitionUri": "blood_screening.json",
      "type": "create"
    },
    {
      "identifier": "2796fba9-1a0a-54a0-8bc4-31f654094614",
      "prefix": 3,
      "title": "Bednet Distribution",
      "description": "Visit 100% of residential structures in the operational area and provide nets",
      "code": "Bednet Distribution",
      "timingPeriod": {
        "start": "2020-08-13",
        "end": "2020-08-20"
      },
      "reason": "Routine",
      "goalId": "RACD_bednet_distribution",
      "subjectCodableConcept": {
        "text": "Location"
      },
      "trigger": [
        {
          "type": "named-event",
          "name": "plan-activation"
        },
        {
          "type": "named-event",
          "name": "event-submission",
          "expression": {
            "description": "Trigger when a Family Registration event is submitted",
            "expression": "questionnaire = 'Family_Registration'"
          }
        }
      ],
      "condition": [
        {
          "kind": "applicability",
          "subjectCodableConcept": {
            "text": "Family"
          },
          "expression": {
            "description": "Structure is residential or type does not exist and Family exists",
            "expression": "$this.is(FHIR.QuestionnaireResponse) or (($this.type.where(id='locationType').exists().not() or $this.type.where(id='locationType').text = 'Residential Structure') and $this.contained.exists())"
          }
        },
        {
          "kind": "applicability",
          "expression": {
            "description": "Register structure event submitted for a residential structure",
            "expression": "$this.is(FHIR.Location)  or (questionnaire = 'Register_Structure' and $this.item.where(linkId='structureType').answer.value ='Residential Structure')"
          }
        }
      ],
      "definitionUri": "bednet_distribution.json",
      "type": "create"
    },
    {
      "identifier": "14e7ba9b-76d0-504a-b098-effe6c146f95",
      "prefix": 4,
      "title": "Larval Dipping",
      "description": "Perform a minimum of three larval dipping activities in the operational area",
      "code": "Larval Dipping",
      "timingPeriod": {
        "start": "2020-08-13",
        "end": "2020-08-20"
      },
      "reason": "Investigation",
      "goalId": "Larval_Dipping",
      "subjectCodableConcept": {
        "text": "Location"
      },
      "trigger": [
        {
          "type": "named-event",
          "name": "plan-activation"
        },
        {
          "type": "named-event",
          "name": "event-submission",
          "expression": {
            "description": "Trigger when a Register_Structure event is submitted",
            "expression": "questionnaire = 'Register_Structure'"
          }
        }
      ],
      "condition": [
        {
          "kind": "applicability",
          "expression": {
            "description": "Structure is a larval breeding site",
            "expression": "$this.is(FHIR.QuestionnaireResponse) or $this.type.where(id='locationType').text = 'Larval Breeding Site'"
          }
        },
        {
          "kind": "applicability",
          "expression": {
            "description": "Apply to larval breeding sites in Register_Structure questionnaires",
            "expression": "$this.is(FHIR.Location) or (questionnaire = 'Register_Structure' and $this.item.where(linkId='structureType').answer.value ='Larval Breeding Site')"
          }
        }
      ],
      "definitionUri": "larval_dipping_form.json",
      "type": "create"
    },
    {
      "identifier": "183a1a40-e517-5e0b-ade1-c4a540056706",
      "prefix": 5,
      "title": "Mosquito Collection",
      "description": "Set a minimum of three mosquito collection traps and complete the mosquito collection process",
      "code": "Mosquito Collection",
      "timingPeriod": {
        "start": "2020-08-13",
        "end": "2020-08-20"
      },
      "reason": "Investigation",
      "goalId": "Mosquito_Collection",
      "subjectCodableConcept": {
        "text": "Location"
      },
      "trigger": [
        {
          "type": "named-event",
          "name": "plan-activation"
        },
        {
          "type": "named-event",
          "name": "event-submission",
          "expression": {
            "description": "Trigger when a Register_Structure event is submitted",
            "expression": "questionnaire = 'Register_Structure'"
          }
        }
      ],
      "condition": [
        {
          "kind": "applicability",
          "expression": {
            "description": "Structure is a mosquito collection point",
            "expression": "$this.is(FHIR.QuestionnaireResponse) or $this.type.where(id='locationType').text = 'Mosquito Collection Point'"
          }
        },
        {
          "kind": "applicability",
          "expression": {
            "description": "Apply to mosquito collection point in Register_Structure questionnaires",
            "expression": "$this.is(FHIR.Location) or (questionnaire = 'Register_Structure' and $this.item.where(linkId='structureType').answer.value ='Mosquito Collection Point')"
          }
        }
      ],
      "definitionUri": "mosquito_collection_form.json",
      "type": "create"
    },
    {
      "identifier": "302b00cc-7817-5663-b7a5-fb053fecc698",
      "prefix": 6,
      "title": "Behaviour Change Communication",
      "description": "Conduct BCC activity",
      "code": "BCC",
      "timingPeriod": {
        "start": "2020-08-13",
        "end": "2020-08-20"
      },
      "reason": "Routine",
      "goalId": "BCC_Focus",
      "subjectCodableConcept": {
        "text": "Jurisdiction"
      },
      "trigger": [
        {
          "type": "named-event",
          "name": "plan-activation"
        }
      ],
      "condition": [
        {
          "kind": "applicability",
          "expression": {
            "description": "Jurisdiction type location",
            "expression": "Location.physicalType.text = 'jdn'"
          }
        }
      ],
      "definitionUri": "behaviour_change_communication.json",
      "type": "create"
    }
  ]
```

### PR DESCRIPTION
**Pull request by ciremusyoka**
Closes #1168 
Matches the dynamic plan template actions to the ones provide on the issue(#1168).

### CODE REVIEW

File: src/components/forms/PlanForm/tests/__snapshots__/index.test.tsx.snap [hunk 1]
```diff
@@ -4556,26 +4539,6 @@ Array [
     required={true}
     value="Apply to residential structures in Register_Structure questionnaires"
   />,
-  <textarea
```
Comment (moshthepitt) [inline src/components/forms/PlanForm/tests/__snapshots__/index.test.tsx.snap:4559]: Why is this being deleted?

File: src/components/forms/PlanForm/tests/__snapshots__/index.test.tsx.snap [hunk 2]
```diff
@@ -4488,30 +4488,13 @@ Array [
...
-    className="col-sm-2 col-form-label"
-    htmlFor="activities[1].condition[0].description"
-  >
-    Description
-  </label>,
 ]
 `;
 
 exports[`containers/forms/PlanForm renders dynamic plans correctly: conditions legends 1`] = `
-Array [
```
Comment (moshthepitt) [inline src/components/forms/PlanForm/tests/__snapshots__/index.test.tsx.snap:4507]: Why is this being deleted?

File: src/components/forms/PlanForm/tests/__snapshots__/index.test.tsx.snap [hunk 3]
```diff
@@ -4488,30 +4488,13 @@ Array [
   >
     Description
   </label>,
-  <label
```
Comment (moshthepitt) [inline src/components/forms/PlanForm/tests/__snapshots__/index.test.tsx.snap:4491]: Why is this being deleted?

File: src/components/forms/PlanForm/tests/__snapshots__/index.test.tsx.snap [hunk 4]
```diff
@@ -4605,18 +4568,6 @@ Array [
     type="text"
     value="event-submission"
   />,
-  <input
```
Comment (moshthepitt) [inline src/components/forms/PlanForm/tests/__snapshots__/index.test.tsx.snap:4608]: Why is this being deleted?

File: src/components/forms/PlanForm/tests/__snapshots__/index.test.tsx.snap [hunk 5]
```diff
@@ -4646,24 +4597,13 @@ Array [
   >
     Description
   </label>,
-  <label
```
Comment (moshthepitt) [inline src/components/forms/PlanForm/tests/__snapshots__/index.test.tsx.snap:4649]: Why is this being deleted?

File: src/components/forms/PlanForm/tests/__snapshots__/index.test.tsx.snap [hunk 6]
```diff
@@ -4646,24 +4597,13 @@ Array [
...
-    className="col-sm-2 col-form-label"
-    htmlFor="activities[1].trigger[0].name"
-  >
-    Name
-  </label>,
 ]
 `;
 
 exports[`containers/forms/PlanForm renders dynamic plans correctly: triggers legends 1`] = `
-Array [
```
Comment (moshthepitt) [inline src/components/forms/PlanForm/tests/__snapshots__/index.test.tsx.snap:4659]: Why is this being deleted?

**Comment by moshthepitt** [comment]
Could you add screenshots of the different plans in a form? @ciremusyoka

File: src/configs/settings.ts [hunk 7]
```diff
@@ -957,25 +952,18 @@ export const planActivities: PlanActivities = {
...
           },
           kind: APPLICABILITY_CONDITION_KIND,
-        },
-        {
-          expression: {
-            description: 'Family does not exist for structure',
-            expression: '$this.contained.exists().not()',
-            subjectCodableConcept: {
-              text: 'Family',
-            },
```
Comment (ciremusyoka) [inline src/configs/settings.ts:970]: @moshthepitt deletion of this action is causing the some deletions on the snapshots for both `Dynamic FI` and `dynamic MDA`.

File: src/components/forms/PlanForm/helpers.ts [hunk 8]
```diff
@@ -245,10 +245,7 @@ export const DynamicMDAActivities = pick(planActivities, [
   DYNAMIC_MDA_COMMUNITY_DISPENSE_ACTIVITY_CODE,
   DYNAMIC_FAMILY_REGISTRATION_ACTIVITY_CODE,
 ]);
-export const DynamicIRSActivities = pick(planActivities, [
-  DYNAMIC_BCC_ACTIVITY_CODE,
-  DYNAMIC_IRS_ACTIVITY_CODE,
-]);
```
Comment (ciremusyoka) [inline src/components/forms/PlanForm/helpers.ts:251]: @moshthepitt also removing `DYNAMIC_BCC_ACTIVITY_CODE` from  here is causing changes on the snapshots.

**Comment by ciremusyoka** [comment]
Dynamic IRS snapshots
Screenshot from 2020-08-20 14-51-28
Screenshot from 2020-08-20 14-51-41

**Comment by ciremusyoka** [comment]
Dynamic MDA screenshots
Screenshot from 2020-08-20 14-55-22
Screenshot from 2020-08-20 14-56-01
Screenshot from 2020-08-20 14-56-16
Screenshot from 2020-08-20 14-56-27
Screenshot from 2020-08-20 14-56-44

**Review by moshthepitt — APPROVED** [review]

**Review by p-netm — APPROVED** [review]

**Pull request merged**